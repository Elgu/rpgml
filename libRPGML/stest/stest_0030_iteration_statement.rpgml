for i = 1, 5, 3
{
  PRINT( i );
  PRINT( " " );
}
PRINT( "\n" );

for i = 0 to 3
{
  PRINT( i );
  PRINT( " " );
}
PRINT( "\n" );

for i = 0 to 4 step 2
{
  PRINT( i );
  PRINT( " " );
}
PRINT( "\n" );

for i = 6 to 3 step -1
{
  PRINT( i );
  PRINT( " " );
}
PRINT( "\n" );

for i = 6 to 3 step -3
{
  PRINT( i );
  PRINT( " " );
}
PRINT( "\n" );

for i in [ 6, 3, 9 ]
{
  PRINT( i );
  PRINT( " " );
}
PRINT( "\n" );

for i in []
{
  PRINT( i );
  PRINT( " " );
}
PRINT( "\n" );

for i in [ 23, 0.75, "foo", Frame{ int i=43; } ]
{
  PRINT( i );
  PRINT( " " );
}
PRINT( "\n" );

for i in Frame{}
{
  PRINT( i );
  PRINT( " " );
}
PRINT( "\n" );

for i in Frame{ double d = 2.125; float f = 1.5; string s = "s"; }
{
  PRINT( i );
  PRINT( " " );
}
PRINT( "\n" );

