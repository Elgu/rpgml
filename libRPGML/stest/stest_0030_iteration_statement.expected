1 5 3
0 1 2 3
0 2 4
6 5 4 3
6 3
6 3 9

23 0.75 foo Frame{ int32 i = 43; }

2.125 1.5 s
